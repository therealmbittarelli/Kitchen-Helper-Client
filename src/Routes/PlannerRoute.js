import React, { Component } from "react";
import { Link } from "react-router-dom";
import config from "../config";
import TokenService from "../Helpers/Token";
import AddMealPlan from "../Components/Planner/MakeMealPlans";
import "../Styles/Planner.css";

class PlannerRoute extends Component {
  constructor(props) {
    super(props);

    this.state = {
      mealplans: [],
      addMealPlan: false
    };
  }

  componentDidMount() {
    this.getMealPlans();
  }

  getMealPlans = () => {
    const url = `${config.API_ENDPOINT}/planner`;
    const authToken = TokenService.getAuthToken();
    fetch(url, {
      method: "GET",
      headers: {
        "Content-type": "application/json",
        Authorization: `Bearer ${authToken}`
      }
    })
      .then(res => res.json())
      .then(data => {
        return this.setState({
          mealplans: data
        });
      });
  };


  renderMealPlans = () => {
    const mealplans = this.state.mealplans;
    return mealplans.map(mealplan => (
      <section className="mealplan-section" key={mealplan.id}>
        <h2 className="mealplan-title">{mealplan.title}</h2>
        <span className="planned-date">Planned for: {mealplan.planned_date}</span>{" "}
        <br />
        <br />
        <Link
          to={{
            pathname: `/planner/${mealplan.id}`,
            state: {
              id: mealplan.id,
              title: mealplan.title,
              planned_date: mealplan.planned_date,
              time_to_make: mealplan.time_to_make,
              needed_ingredients: mealplan.needed_ingredients
            }
          }}
        >
          <button className="smallButton">
            View/Edit Meal Plan</button>
        </Link>
        <br />
      </section>
    ));
  };

  setStateAddMealPlanTrue = () => {
    this.setState({ addMealPlan: true });
  };

  setStateAddMealPlanFalse = () => {
    this.setState({ addMealPlan: false });
  };

  render() {
    return (
      <section id="planner-route-container">
        <h2 className="my-planner-header" id="my-planner-header">
          My Meal Plans
        </h2>
        <p id="planner-text">
          Create, update, and manage your meal plans
        </p>
        <button
          className="bigButton"
          type="submit"
          onClick={() => this.setStateAddMealPlanTrue()}
        >
          Add a Meal Plan
        </button>
        <AddMealPlan
          addMealPlan={this.state.addMealPlan}
          allMealPlans={this.state.mealplans}
          refreshMealPlans={this.getMealPlans}
          closeAddForm={this.setStateAddMealPlanFalse}
        />
        {this.state.mealplans && this.renderMealPlans()}
      </section>
    );
  }
}

export default PlannerRoute;
