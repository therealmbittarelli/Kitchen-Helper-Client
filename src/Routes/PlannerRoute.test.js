import React from 'react';
import ReactDOM from 'react-dom';
import PlannerRoute from './PlannerRoute';
import { BrowserRouter } from 'react-router-dom';
import renderer from "react-test-renderer";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<BrowserRouter>
    <PlannerRoute />
  </BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer.create(<BrowserRouter>
    <PlannerRoute />
  </BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});