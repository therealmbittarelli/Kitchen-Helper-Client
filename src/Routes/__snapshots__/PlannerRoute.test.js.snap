// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  id="planner-route-container"
>
  <h2
    className="my-planner-header"
    id="my-planner-header"
  >
    My Meal Plans
  </h2>
  <p
    id="planner-text"
  >
    Create, update, and manage your meal plans
  </p>
  <button
    className="bigButton"
    onClick={[Function]}
    type="submit"
  >
    Add a Meal Plan
  </button>
  <div />
</section>
`;
